### PR TITLE
Adds ability to upload scene assets to SCNetworking

### PR DESCRIPTION
**Pull request by jeffchuber**
Not a common use case, but some customer want to attach arbitrary data to a scene. This enables that use case.

### CODE REVIEW

File: StandardCyborgNetworking/StandardCyborgNetworking/Operations/ServerSceneOperations.swift [hunk 1]
```diff
@@ -194,6 +195,112 @@ public class ServerAddSceneOperation: ServerOperation {
...
+public class ServerAddSceneAssetOperation: ServerOperation {
+    let fileAssetURL: URL
+    let teamKey: String?
+    let sceneKey: String?
+    let remoteFileName: String?
+    
+    public init(fileAssetURL: URL,
+                teamKey: String? = nil,
+                sceneKey: String? = nil,
+                remoteFileName: String? = nil,
```
Comment (rreusser) [inline StandardCyborgNetworking/StandardCyborgNetworking/Operations/ServerSceneOperations.swift]:
```suggestion
                remoteFileName: String? = nil,
```

File: StandardCyborgNetworking/StandardCyborgNetworking/Model/ServerScene.swift [hunk 2]
```diff
@@ -123,6 +123,49 @@ public struct ServerSceneVersion: Codable {
     }
 }
 
+public struct ServerSceneAsset: Codable {
+    
+    private enum CodingKeys: String, CodingKey {
+        case createdAt
```
Comment (rreusser) [inline StandardCyborgNetworking/StandardCyborgNetworking/Model/ServerScene.swift:129]: The camel casing vs. underscore casing is an unfortunate thing. I don't mind as long as things are consistent and work.

Comment (rreusser) [reply StandardCyborgNetworking/StandardCyborgNetworking/Model/ServerScene.swift:129]: Related: should this be `created_at`?

File: StandardCyborgNetworking/StandardCyborgNetworking/Operations/ServerSceneOperations.swift [hunk 3]
```diff
@@ -194,6 +195,112 @@ public class ServerAddSceneOperation: ServerOperation {
...
+        var sceneAssetUploadInfo: S3UploadInfo!
+        
+        //1. fetch scene by key to make sure it's valid
+        //2. create direct upload
+        //3. upload to direct upload
+        //4. post to scenes/:uid/assets with direct upload key and scene uid to link as asset
+        
+        var promise =
+        firstly {
+            // 4. Upload files to S3 via the POST `/direct_uploads` endpoint
```
Comment (rreusser) [inline StandardCyborgNetworking/StandardCyborgNetworking/Operations/ServerSceneOperations.swift]: is `4.` a typo/holdover here?

**Review by rreusser — APPROVED** [review]
It looks pretty good to me, and 2.1.0 seems to make sense, *as long as* it's not intended in any way to correspond with StandardCyborgSDK versioning, which also happens to currently be at 2.1.0.

File: StandardCyborgNetworking/StandardCyborgNetworking/Model/ServerScene.swift (hunk 2 above)
Comment (jeffchuber) [reply StandardCyborgNetworking/StandardCyborgNetworking/Model/ServerScene.swift:129]: I wanted to keep it consistent with `ServerSceneVersion`.

File: StandardCyborgNetworking/StandardCyborgNetworking/Operations/ServerSceneOperations.swift (hunk 3 above)
Comment (jeffchuber) [reply StandardCyborgNetworking/StandardCyborgNetworking/Operations/ServerSceneOperations.swift]: cleaned up

**Pull request merged**